ttnn/cpp/ttnn/operations/eltwise/ternary/where_op.cpp

```diff
@@ -0,0 +1,85 @@
+        output_tensor = ttnn::add(queue_id, t2, t1, std::nullopt, output_mem_config);
+    }
+    return output_tensor.value();
+}
+
+Tensor WhereOp::_where_v1(
+    uint8_t queue_id, const Tensor& predicate, const float value_true, const Tensor& value_false, const std::optional<MemoryConfig>& output_mem_config, std::optional<Tensor> output_tensor) {
+
+    Tensor t2 = ttnn::multiply(queue_id, ttnn::gtz(queue_id, predicate, output_mem_config), value_true, std::nullopt, output_mem_config);
+
+    if(output_tensor.has_value()){
+        ttnn::multiply(queue_id, ttnn::lez(queue_id, predicate, output_mem_config), value_false, std::nullopt, operation::DEFAULT_OUTPUT_MEMORY_CONFIG , output_tensor);
+        ttnn::add(queue_id, t2, output_tensor.value(), std::nullopt, operation::DEFAULT_OUTPUT_MEMORY_CONFIG, output_tensor);
+    }
+    else
+    {
+        Tensor t1 = ttnn::multiply(queue_id, ttnn::lez(queue_id, predicate, output_mem_config), value_false, std::nullopt, output_mem_config);
+        output_tensor = ttnn::add(queue_id, t2, t1, std::nullopt, output_mem_config);
+    }
+    return output_tensor.value();
+}
+
+Tensor WhereOp::_where_v2(
+    uint8_t queue_id, const Tensor& predicate, const Tensor& value_true, float value_false, const std::optional<MemoryConfig>& output_mem_config, std::optional<Tensor> output_tensor) {
+
+    Tensor t1 = ttnn::multiply(queue_id, ttnn::lez(queue_id, predicate, output_mem_config), value_false, std::nullopt, output_mem_config);
+
+    if(output_tensor.has_value()){
+        ttnn::multiply(queue_id, ttnn::gtz(queue_id, predicate, output_mem_config), value_true, std::nullopt, operation::DEFAULT_OUTPUT_MEMORY_CONFIG, output_tensor);
+        ttnn::add(queue_id, output_tensor.value(), t1, std::nullopt, operation::DEFAULT_OUTPUT_MEMORY_CONFIG, output_tensor);
+    }
+    else
+    {
+        Tensor t2 = ttnn::multiply(queue_id, ttnn::gtz(queue_id, predicate, output_mem_config), value_true, std::nullopt, output_mem_config);
+        output_tensor = ttnn::add(queue_id, t2, t1, std::nullopt, output_mem_config);
+    }
+    return output_tensor.value();
+}
+
+Tensor WhereOp::_where_v3(
+    uint8_t queue_id, const Tensor& predicate, const float value_true, const float value_false, const std::optional<MemoryConfig>& output_mem_config, std::optional<Tensor> output_tensor) {
+    Tensor t2 = ttnn::multiply(queue_id, ttnn::gtz(queue_id, predicate, output_mem_config), value_true, std::nullopt, output_mem_config);
+    Tensor t1 = ttnn::multiply(queue_id, ttnn::lez(queue_id, predicate, output_mem_config), value_false, std::nullopt, output_mem_config);
+    if(output_tensor.has_value()){
+        ttnn::add(queue_id, t2, t1, std::nullopt, operation::DEFAULT_OUTPUT_MEMORY_CONFIG, output_tensor);
+    } else {
+        output_tensor = ttnn::add(queue_id, t2, t1, std::nullopt, output_mem_config);
+    }
+    return output_tensor.value();
+}
```

Comment:
Is there a way to avoid duplication like this?

Reply:
@TT-BrianLiu yes, I think we now can. 
previously there were two api's mul( tensor * tensor) and mul_unary( tensor * scalar) so these variations had to be implemented separately. 
Recently we combined both functionalities available under one api for ttnn.multiply, we can go for a single implementation code. I will update these.

Reply:
I have added the changes